lib/ReactViews/Generic/Editor.tsx

```diff
@@ -21,18 +22,34 @@ import "tinymce/plugins/table";
 import "tinymce/plugins/autolink";
 
 // Extra css to enable proper behaviour of tinymce, including image resize handles
+// @ts-expect-error: raw-loader is unknown to typescript
 import contentCss from "!!raw-loader!tinymce/skins/content/default/content.min.css";
+// @ts-expect-error: raw-loader is unknown to typescript
 import contentUiCss from "!!raw-loader!tinymce/skins/ui/oxide/content.min.css";
 
-export default function TinyEditor(props) {
-  const editorRef = useRef(null);
+interface ITinyEditorProps {
+  html: string;
+  onChange: (html: string) => void;
+  language: string;
+  baseUrl: string;
+}
+
+export default function TinyEditor({
+  html,
+  onChange,
+  language
+}: ITinyEditorProps) {
+  const editorRef = useRef<TinyMCEEditor | null>(null);
 
   return (
     <Editor
+      key={language}
```

Comment:
TinyMCE doesn't support dynamic updates of translations after the editor is initialized, so we need to remount it